### ISSUE TITLE
Getting BELU score 0 in validation set

### ISSUE THREAD

**Issue by canamika27** [h2oai/h2o-llmstudio#56]
Hi Team, I was trying to train pythia-1.4b-deduped using cli & while training I am getting BELU as 0 for validation set, so basically its not able to generate any text. FYI - I followed the same tutorial shared in the colab notebook of repo

I have shared the logs below, can you please tell if I am going wrong somewhere ?

CUDA_SETUP: WARNING! libcudart.so not found in any environmental path. Searching /usr/local/cuda/lib64...
CUDA SETUP: CUDA runtime path found: /usr/local/cuda/lib64/libcudart.so.11.0
CUDA SETUP: Highest compute capability among GPUs detected: 8.0
CUDA SETUP: Detected CUDA version 117
CUDA SETUP: Loading binary /home/dell/.conda/envs/h2oai/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda117.so...
2023-04-26 17:53:58,234 - INFO: Global random seed: 1
2023-04-26 17:53:58,234 - INFO: Preparing the data...
2023-04-26 17:53:58,234 - INFO: Setting up automatic validation split...
2023-04-26 17:53:58,376 - INFO: Preparing train and validation data
2023-04-26 17:53:58,377 - INFO: Loading train dataset...
Using pad_token, but it is not set yet.
Using cls_token, but it is not set yet.
Using sep_token, but it is not set yet.
2023-04-26 17:53:58,506 - INFO: Stop token ids: [tensor([])]
2023-04-26 17:53:58,563 - INFO: Sample prompt: As an AI what are your personal feelings on Sarah Connor?<|endoftext|>
2023-04-26 17:53:58,563 - INFO: Loading validation dataset...
Using pad_token, but it is not set yet.
Using cls_token, but it is not set yet.
Using sep_token, but it is not set yet.
2023-04-26 17:53:58,677 - INFO: Stop token ids: [tensor([])]
2023-04-26 17:53:58,681 - INFO: Sample prompt: What types of tests do we have in software development?<|endoftext|>
2023-04-26 17:53:58,681 - INFO: Number of observations in train dataset: 8191
2023-04-26 17:53:58,681 - INFO: Number of observations in validation dataset: 83
2023-04-26 17:53:58,688 - INFO: dtype: torch.float16
/home/dell/.conda/envs/h2oai/lib/python3.10/site-packages/peft/tuners/lora.py:173: UserWarning: fan_in_fan_out is set to True but the target module is not a Conv1D. Setting fan_in_fan_out to False.
  warnings.warn(
trainable params: 786432 || all params: 1415434240 || trainable%: 0.055561182411413196
2023-04-26 17:54:05,371 - INFO: Training Epoch: 1 / 1
2023-04-26 17:54:05,372 - INFO: train loss:   0%|          | 0/2047 [00:00<?, ?it/s]
Using pad_token, but it is not set yet.
Using cls_token, but it is not set yet.
Using sep_token, but it is not set yet.
2023-04-26 17:54:05,779 - INFO: Stop token ids: [tensor([])]
2023-04-26 17:55:01,890 - INFO: train loss: 2.22:   5%|4         | 102/2047 [00:56<17:57,  1.80it/s]
2023-04-26 17:55:15,374 - INFO: train loss: 2.22:   5%|4         | 102/2047 [01:10<17:57,  1.80it/s]
2023-04-26 17:55:42,947 - INFO: train loss: 2.21:  10%|9         | 204/2047 [01:37<14:16,  2.15it/s]
2023-04-26 17:55:55,376 - INFO: train loss: 2.21:  10%|9         | 204/2047 [01:50<14:16,  2.15it/s]
2023-04-26 17:56:18,324 - INFO: train loss: 2.17:  15%|#4        | 306/2047 [02:12<11:55,  2.43it/s]
2023-04-26 17:56:35,377 - INFO: train loss: 2.17:  15%|#4        | 306/2047 [02:30<11:55,  2.43it/s]
2023-04-26 17:56:55,301 - INFO: train loss: 1.91:  20%|#9        | 408/2047 [02:49<10:42,  2.55it/s]
2023-04-26 17:57:05,378 - INFO: train loss: 1.91:  20%|#9        | 408/2047 [03:00<10:42,  2.55it/s]
2023-04-26 17:57:30,784 - INFO: train loss: 2.27:  25%|##4       | 510/2047 [03:25<09:37,  2.66it/s]
2023-04-26 17:57:45,380 - INFO: train loss: 2.27:  25%|##4       | 510/2047 [03:40<09:37,  2.66it/s]
2023-04-26 17:58:06,714 - INFO: train loss: 2.04:  30%|##9       | 612/2047 [04:01<08:47,  2.72it/s]
2023-04-26 17:58:25,381 - INFO: train loss: 2.04:  30%|##9       | 612/2047 [04:20<08:47,  2.72it/s]
2023-04-26 17:58:41,145 - INFO: train loss: 2.22:  35%|###4      | 714/2047 [04:35<07:57,  2.79it/s]
2023-04-26 17:58:55,383 - INFO: train loss: 2.22:  35%|###4      | 714/2047 [04:50<07:57,  2.79it/s]
2023-04-26 17:59:16,807 - INFO: train loss: 2.03:  40%|###9      | 816/2047 [05:11<07:17,  2.81it/s]
2023-04-26 17:59:35,384 - INFO: train loss: 2.03:  40%|###9      | 816/2047 [05:30<07:17,  2.81it/s]
2023-04-26 17:59:52,680 - INFO: train loss: 1.96:  45%|####4     | 918/2047 [05:47<06:39,  2.82it/s]
2023-04-26 18:00:05,386 - INFO: train loss: 1.96:  45%|####4     | 918/2047 [06:00<06:39,  2.82it/s]
2023-04-26 18:00:33,621 - INFO: train loss: 2.27:  50%|####9     | 1020/2047 [06:28<06:18,  2.71it/s]
2023-04-26 18:00:45,388 - INFO: train loss: 2.27:  50%|####9     | 1020/2047 [06:40<06:18,  2.71it/s]
2023-04-26 18:01:12,389 - INFO: train loss: 2.12:  55%|#####4    | 1122/2047 [07:07<05:44,  2.69it/s]
2023-04-26 18:01:25,389 - INFO: train loss: 2.12:  55%|#####4    | 1122/2047 [07:20<05:44,  2.69it/s]
2023-04-26 18:01:53,475 - INFO: train loss: 2.11:  60%|#####9    | 1224/2047 [07:48<05:13,  2.62it/s]
2023-04-26 18:02:05,391 - INFO: train loss: 2.11:  60%|#####9    | 1224/2047 [08:00<05:13,  2.62it/s]
2023-04-26 18:02:31,822 - INFO: train loss: 2.00:  65%|######4   | 1326/2047 [08:26<04:33,  2.63it/s]
2023-04-26 18:02:45,393 - INFO: train loss: 2.00:  65%|######4   | 1326/2047 [08:40<04:33,  2.63it/s]
2023-04-26 18:03:07,590 - INFO: train loss: 2.02:  70%|######9   | 1428/2047 [09:02<03:49,  2.70it/s]
2023-04-26 18:03:25,394 - INFO: train loss: 2.02:  70%|######9   | 1428/2047 [09:20<03:49,  2.70it/s]
2023-04-26 18:03:47,965 - INFO: train loss: 2.08:  75%|#######4  | 1530/2047 [09:42<03:15,  2.64it/s]
2023-04-26 18:04:05,396 - INFO: train loss: 2.08:  75%|#######4  | 1530/2047 [10:00<03:15,  2.64it/s]
2023-04-26 18:04:23,403 - INFO: train loss: 2.01:  80%|#######9  | 1632/2047 [10:18<02:33,  2.71it/s]
2023-04-26 18:04:35,397 - INFO: train loss: 2.01:  80%|#######9  | 1632/2047 [10:30<02:33,  2.71it/s]
2023-04-26 18:04:59,964 - INFO: train loss: 2.11:  85%|########4 | 1734/2047 [10:54<01:54,  2.73it/s]
2023-04-26 18:05:15,399 - INFO: train loss: 2.11:  85%|########4 | 1734/2047 [11:10<01:54,  2.73it/s]
2023-04-26 18:05:39,237 - INFO: train loss: 2.09:  90%|########9 | 1836/2047 [11:33<01:18,  2.69it/s]
2023-04-26 18:05:55,401 - INFO: train loss: 2.09:  90%|########9 | 1836/2047 [11:50<01:18,  2.69it/s]
2023-04-26 18:06:16,531 - INFO: train loss: 2.05:  95%|#########4| 1938/2047 [12:11<00:40,  2.70it/s]
2023-04-26 18:06:35,402 - INFO: train loss: 2.05:  95%|#########4| 1938/2047 [12:30<00:40,  2.70it/s]
2023-04-26 18:06:55,230 - INFO: train loss: 2.00: 100%|#########9| 2040/2047 [12:49<00:02,  2.68it/s]
2023-04-26 18:06:57,568 - INFO: train loss: 1.93: 100%|##########| 2047/2047 [12:52<00:00,  2.65it/s]
2023-04-26 18:06:57,570 - INFO: Starting validation inference
2023-04-26 18:06:57,570 - INFO: validation progress:   0%|          | 0/21 [00:00<?, ?it/s]
2023-04-26 18:06:58,259 - INFO: validation progress:   5%|4         | 1/21 [00:00<00:13,  1.45it/s]
2023-04-26 18:06:58,528 - INFO: validation progress:  10%|9         | 2/21 [00:00<00:08,  2.26it/s]
2023-04-26 18:06:58,768 - INFO: validation progress:  14%|#4        | 3/21 [00:01<00:06,  2.86it/s]
2023-04-26 18:06:58,991 - INFO: validation progress:  19%|#9        | 4/21 [00:01<00:05,  3.34it/s]
2023-04-26 18:06:59,203 - INFO: validation progress:  24%|##3       | 5/21 [00:01<00:04,  3.73it/s]
2023-04-26 18:06:59,507 - INFO: validation progress:  29%|##8       | 6/21 [00:01<00:04,  3.57it/s]
2023-04-26 18:06:59,732 - INFO: validation progress:  33%|###3      | 7/21 [00:02<00:03,  3.81it/s]
2023-04-26 18:06:59,918 - INFO: validation progress:  38%|###8      | 8/21 [00:02<00:03,  4.20it/s]
2023-04-26 18:07:00,148 - INFO: validation progress:  43%|####2     | 9/21 [00:02<00:02,  4.25it/s]
2023-04-26 18:07:00,367 - INFO: validation progress:  48%|####7     | 10/21 [00:02<00:02,  4.34it/s]
2023-04-26 18:07:00,607 - INFO: validation progress:  52%|#####2    | 11/21 [00:03<00:02,  4.29it/s]
2023-04-26 18:07:00,818 - INFO: validation progress:  57%|#####7    | 12/21 [00:03<00:02,  4.42it/s]
2023-04-26 18:07:01,090 - INFO: validation progress:  62%|######1   | 13/21 [00:03<00:01,  4.16it/s]
2023-04-26 18:07:01,353 - INFO: validation progress:  67%|######6   | 14/21 [00:03<00:01,  4.04it/s]
2023-04-26 18:07:01,541 - INFO: validation progress:  71%|#######1  | 15/21 [00:03<00:01,  4.36it/s]
2023-04-26 18:07:01,781 - INFO: validation progress:  76%|#######6  | 16/21 [00:04<00:01,  4.30it/s]
2023-04-26 18:07:02,006 - INFO: validation progress:  81%|########  | 17/21 [00:04<00:00,  4.34it/s]
2023-04-26 18:07:02,221 - INFO: validation progress:  86%|########5 | 18/21 [00:04<00:00,  4.43it/s]
2023-04-26 18:07:02,434 - INFO: validation progress:  90%|######### | 19/21 [00:04<00:00,  4.51it/s]
2023-04-26 18:07:02,800 - INFO: validation progress:  95%|#########5| 20/21 [00:05<00:00,  3.77it/s]
2023-04-26 18:07:03,217 - INFO: validation progress: 100%|##########| 21/21 [00:05<00:00,  3.22it/s]
2023-04-26 18:07:03,269 - INFO: validation progress: 100%|##########| 21/21 [00:05<00:00,  3.69it/s]
2023-04-26 18:07:03,449 - INFO: Mean validation loss: 2.13012
2023-04-26 18:07:03,520 - INFO: Validation BLEU: 0.00000
2023-04-26 18:07:03,891 - INFO: Saving last model checkpoint: val_loss 2.1301, val_BLEU 0.0 to output/demo_oasst-data/
all done

**Comment by pascal-pfeiffer** [comment]
Thank you for reporting @canamika27 

Actually, I believe you ran into a bug with stop tokens cutting off the prediction at the first token. 
It was fixed in this PR: https://github.com/h2oai/h2o-llmstudio/pull/54

Can you check if the issue still persists on the current main branch, please.

**Comment by canamika27** [comment]
Its working now Thanks !!
